### ISSUE TITLE
'not' operator precedence displayed wrong

### ISSUE THREAD

**Issue by rayanamal** [nushell/nushell#13675]
### Describe the bug

Operators and their precedence values are given with the `help operators` command:
![image](https://github.com/user-attachments/assets/a48aa975-6cb8-4706-86fc-7e1e147a49a7)

`not` operator precedence is given as 0. `or` precedence is 40.

---
```nu
~> if not (false) or (true) { print hello }
```

This prints `hello`. If `or` precedence was higher than `not`, it wouldn't print anything, since the evaluation order would be like the following:
```nu
~> if not ((false) or (true)) { print hello }
```

### How to reproduce

Run the given command.

### Expected behavior

`help operators` command to display the correct precedence.

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.96.1                                                            |
| major              | 0                                                                 |
| minor              | 96                                                                |
| patch              | 1                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06) (built from a source tarball) |
| cargo_version      | cargo 1.80.0 (376290515 2024-07-16)                               |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash                          |
| installed_plugins  |                                                                   |

### Additional context

I feel there is a special reason I don't know for why `not`s precedence is 0 here.

**Comment by sholderbach** [comment]
Good catch. At the same time our handling of `not` (as the only unary operator) is kind of weird (due to how it may/may not behave similar to a command) so it may not have a consistent precedence, so worth a little digging.
